File: conversions/typescript/Construct.ts
import { BaseElement } from './base_elements';

export interface Construct extends BaseElement {
  // Nature
  history?: string | null;
  understanding?: string | null;
  state?: string | null;
  startDate?: number | null;
  endDate?: number | null;
  founderId?: string | null;
  organiserId?: string | null;
  // Involves
  charactersIds?: string[] | null;
  objectsIds?: string[] | null;
  locationsIds?: string[] | null;
  speciesIds?: string[] | null;
  creaturesIds?: string[] | null;
  institutionsIds?: string[] | null;
  traitsIds?: string[] | null;
  collectivesIds?: string[] | null;
  territoriesIds?: string[] | null;
  abilitiesIds?: string[] | null;
  phenomenaIds?: string[] | null;
  languagesIds?: string[] | null;
  familiesIds?: string[] | null;
  relationsIds?: string[] | null;
  titlesIds?: string[] | null;
  constructsIds?: string[] | null;
}

File: languages/csharp_unity/Construct.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Construct : BaseElement
{
    // Nature
    public string Rationale;
    public string History;
    public string Status;
    public string Reach;
    public int? StartDate;
    public int? EndDate;
    public string FounderId;
    public string CustodianId;
    // Involves
    public List<string> CharactersIds; = new List<string>();
    public List<string> ObjectsIds; = new List<string>();
    public List<string> LocationsIds; = new List<string>();
    public List<string> SpeciesIds; = new List<string>();
    public List<string> CreaturesIds; = new List<string>();
    public List<string> InstitutionsIds; = new List<string>();
    public List<string> TraitsIds; = new List<string>();
    public List<string> CollectivesIds; = new List<string>();
    public List<string> ZonesIds; = new List<string>();
    public List<string> AbilitiesIds; = new List<string>();
    public List<string> PhenomenaIds; = new List<string>();
    public List<string> LanguagesIds; = new List<string>();
    public List<string> FamiliesIds; = new List<string>();
    public List<string> RelationsIds; = new List<string>();
    public List<string> TitlesIds; = new List<string>();
    public List<string> ConstructsIds; = new List<string>();
    public List<string> EventsIds; = new List<string>();
    public List<string> NarrativesIds; = new List<string>();
}
